test/scripts/e2e_subs/min_balance.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+from datetime import datetime
+from pathlib import PurePath
+import sys
+
+import algosdk.future.transaction as txn
+
+from goal import Goal, AtomicABI
+
+CONSENSUS_MIN_BALANCE = 100_000
+ASSET_MIN_BALANCE = 100_000
+APP_MIN_BALANCE = 100_000
+OPTIN_MIN_BALANCE = 100_000
+# app schemas:
+APP_KV_MIN_BALANCE = 25_000
+APP_INTS_MIN_BALANCE = 3_500
+APP_BYTES_MIN_BALANCE = 25_000
+
+# per userBalance.go::MinBalance() as of Dec 2021:
+EXTRA_PAGE_MIN_BALANCE = APP_MIN_BALANCE
+
+
+# Set INTERACTIVE True if you want to run a remote debugger interactively on the given PORT
+INTERACTIVE, DEBUGPORT = False, 4312
+
+
+def initialize_debugger():
+    import multiprocessing
+
+    if multiprocessing.current_process().pid > 1:
+        import debugpy
+
+        debugpy.listen(("0.0.0.0", DEBUGPORT))
+        print("Debugger is ready to be attached, press F5", flush=True)
+        debugpy.wait_for_client()
+        print("Visual Studio Code debugger is now attached", flush=True)
+
+
+if INTERACTIVE:
+    initialize_debugger()
+
+
+def get_pysdk_min_balance(goal, account):
+    return goal.algod.account_info(account)["min-balance"]
+
+
+def get_teal_min_balance(abi, account):
```

Comment:
not sure I understand why do we need all this python abi + teal machinery to test min balance field in REST

Reply:
I agree with @algorandskiy about that one. This python testing should go to the python repository.
Testing the REST endpoint of the algod can easily be done using either expect or one of the existing e2e subs.
( you've already implemented min_balance.sh, which is great )

Reply:
@algonautshant and @tsachiherman  - please have a look again. I've implemented the decision points from yesterday's meeting:

1. Remove dependence on Python SDK's ABI interaction implementation, and in particular, removed the `AtomicABI` test class
2. Keeping the python tests, but refactored to avoid ABI
3. Pinned the Python SDK and added a comment of explanation